src/js/protocols/stm32usbdfu.js

```diff
@@ -149,17 +148,19 @@ STM32DFU_protocol.prototype.claimInterface = function (interfaceNumber) {
         // Don't perform the error check on MacOS at this time as there seems to be a bug
         // where it always reports the Chrome error "Error claiming interface." even though
         // the interface is in fact successfully claimed.
-        if (checkChromeRuntimeError() && (GUI.operating_system !== "MacOS")) {
-            console.log('Failed to claim USB device!');
+        if (checkChromeRuntimeError()) {
+            GUI.log('running cleanup, failed to claim USB interface');
+            console.log(`running cleanup Failed to claim USB interfaceNo: ${interfaceNumber} handle ${self.handle.handle}`);
             self.cleanup();
-        }
-
-        console.log(`Claimed interface: ${interfaceNumber}`);
-
-        if (self.options.exitDfu) {
-            self.leave();
         } else {
-            self.upload_procedure(0);
+            GUI.log('Claimed USB interface - able to progress');
+            console.log(`Claimed interface able to progress interfaceNo: ${interfaceNumber} handle ${self.handle.handle}`);
+            if (self.options.exitDfu) {
```

Comment:
I don't know if it makes sense to put these things here, but they seem conditional on the presence of an active interface, and it works.